webapp/src/Service/ImportExportService.php

```diff
@@ -329,8 +330,20 @@ public function importProblemsData(Contest $contest, array $problems, array &$id
                 ->setTimelimit($problemData['time_limit'] ?? 10)
                 ->setExternalid($problemData['id'] ?? $problemData['short-name'] ?? $problemLabel ?? null);
 
-            $this->em->persist($problem);
-            $this->em->flush();
+            $hasErrors = false;
+            $errors    = $this->validator->validate($problem);
+            if ($errors->count()) {
+                $hasErrors = true;
```

Comment:
```suggestion
            if ($hasErrors = $errors->count()) {
```

Not sure if we like this,

Reply:
I don't really like it, especially if `problem` HAS errors and contestproblem does not.

Reply:
I'd do it like:
```
           $errors    = $this->validator->validate($problem);
           $hasErrors = $errors->count();
           if ($hasErrors) {
```
although I see that `$hasErrors` is again set further down. But then maybe from linebreaks or name make it clear that this variable has more global scope, while `$errors` is just reused.
Will leave it up to you.

Reply:
I used two separate variables, what about that idea? It makes it a bit more readable.